torch_xla/csrc/ops/nll_loss2d_backward.cpp

```diff
@@ -0,0 +1,96 @@
+#include "torch_xla/csrc/ops/nll_loss2d_backward.h"
+
+#include "tensorflow/compiler/xla/xla_client/debug_macros.h"
+#include "tensorflow/compiler/xla/xla_client/util.h"
+#include "torch_xla/csrc/lowering_context.h"
+#include "torch_xla/csrc/nll_loss.h"
+#include "torch_xla/csrc/ops/infer_output_shape.h"
+
+namespace torch_xla {
+namespace ir {
+namespace ops {
+namespace {
+
+xla::Shape NodeOutputShape(const Value& grad_output, const Value& logits,
+                           const Value& labels,
+                           const absl::optional<Value>& weight,
+                           const absl::optional<Value>& total_weight,
+                           ReductionMode reduction, int ignore_index) {
+  auto lower_for_shape_fn =
+      [&](absl::Span<const xla::XlaOp> operands) -> xla::XlaOp {
+    absl::optional<xla::XlaOp> weight;
+    absl::optional<xla::XlaOp> total_weight;
+    if (operands.size() > 3) {
+      XLA_CHECK_EQ(operands.size(), 5)
+          << "If weight is specified, so must be total_weight";
+      weight = operands[3];
+      total_weight = operands[4];
+    }
+    return BuildNllLossBackward(operands[0], operands[1], operands[2], weight,
+                                total_weight, ignore_index, reduction);
+  };
+  std::vector<xla::Shape> shapes;
+  for (auto& input : xla::util::GetValuesVector<Value>(
+           {grad_output, logits, labels}, {&weight, &total_weight})) {
+    shapes.push_back(input.shape());
+  }
+  return InferOutputShape(shapes, lower_for_shape_fn);
+}
+
+}  // namespace
+
+NllLoss2dBackward::NllLoss2dBackward(const Value& grad_output,
+                                     const Value& logits, const Value& labels,
+                                     const absl::optional<Value>& weight,
+                                     const absl::optional<Value>& total_weight,
+                                     ReductionMode reduction, int ignore_index)
+    : Node(ir::OpKind(at::aten::nll_loss_backward),
```

Comment:
Doesn't it have its own string ID?
If not, we need to define one in `xla_ops.*`.

Reply:
typo, fixed